core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableSchema.java

```diff
@@ -21,14 +21,17 @@
 {
     private final SchemaTableName table;
     private final List<ColumnSchema> columns;
+    private final List<String> checkConstraints;
```

Comment:
Do we need this here? Why?

Reply:
To use from `StatementAnalyzer`. I would add a new method in `Metadata` if it's better than here. Or are you suggesting moving to the under column?

Reply:
Feels like not belonging here at all. I'd like to understand this more.

Reply:
Feels like it needs to stay here (or we would have to remove almost all usages of getTableSchema and migrate them to getTableMetadata)